### DISCUSSION TITLE
Avoid Route /favicon.ico doesn't exist

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#20398] [Feature Requests]
### Summary

How to avoid this message with localhost:8055
favicon.ico  is present inside node_modules\@directus\app
I tried to put it to other places but still get this error message.

### Motivation

Get no message error at all

### Detailed Design

- 

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

None

### Alternatives

None

### Adoption Strategy

None

**Comment by TLMondo** [comment]
same here

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by nos3b3ar** [comment]
Hello,
"there's enough community interest"? seriously? Sorry, but this looks very unprofessional.
I mean, you just start with an new blank directus project and first think you see is an ROUTE_NOT_FOUND Error because of an favicon! It's embarrassing...
First I thought its not a big think, I just need to put the favicon on the right place, but it isn't so easy for directus newcomers.
Please fix it!!!!
BTW: Thank you for Directus :-)

**Reply by br41nslug** [reply]
> First I thought its not a big think, I just need to put the favicon on the right place, but it isn't so easy for directus newcomers.

Adding a favicon can be done in `settings > appearance > public favicon`

https://github.com/user-attachments/assets/bda732ea-f639-4461-aade-f37ac5f38c82

**Reply by nos3b3ar** [reply]
Hi,
thank you for the instruction. This reduces the requests to the favicon-Route, but don't fix the Problem. If you surf to an endpoint page the message "DEBUG: Route /favicon.ico doesn't exist." still occurs.

But I fixed it now by creating an endpoint-extension namend "favicon-endpoint" ;)

`export default {
	id: 'favicon.ico',
	handler: (router) => {
		router.get('/', (req, res) => res.send('PASTE YOUR FAVICON- OR SVG-DATA HERE'));
	},
};`

This fix it overall and returns the favicon you want. :o)

**Comment by TLMondo** [comment]
So it might be good to limit the file selector to `.ico` file only, and perhaps some pre-processing to ensure its an `.ico` if dragged in, and that its 32x32. I had unwittingly clicked on a png and was getting the same 404. 

Can confirm it does indeed work when the correct file is set.

**Reply by br41nslug** [reply]
It's not strictly limited to `.ico` files, in my video example im using a SVG favicon and i believe PNG would work too if its the right size.

**Reply by rijkvanzanten** [reply]
People are getting confused by the browser auto-attempting to load `/favicon.ico` and assuming that's a Directus bug or file requirement. It's a similar confusing people have had in the past about the "Error" thrown from trying `/auth/refresh`, even though that's on purpose 🙂

**Reply by TLMondo** [reply]
> It's not strictly limited to `.ico` files, in my video example im using a SVG favicon and i believe PNG would work too if its the right size.

Fair enough. I guess I still vote for feedback to say its the wrong dimension. 

> People are getting confused by the browser auto-attempting to load /favicon.ico and assuming that's a Directus bug or file requirement. It's a similar confusing people have had in the past about the "Error" thrown from trying /auth/refresh, even though that's on purpose 🙂

I think the original intention of the ticket was that it wasn't clear how to set it before the UI option (did that exist at the time?). I think it just needs documentation to address.
